cli/src/git_util.rs

```diff
@@ -484,7 +484,7 @@ pub fn git_fetch(
                     .any(|pattern| pattern.as_exact().is_some_and(|s| s.contains('*')))
                 {
                     user_error_with_hint(
-                        err,
+                        GitFetchError::StarInBranchName,
```

Comment:
I realized there's no reason for this to be a `GitFetchError`, which could simplify the commit. However, I don't know what to use instead. I'll look what happens in other similar situations tomorrow, unless somebody lets me know.